README.md

```diff
@@ -1 +1,1380 @@
+
+    ### <a id="summary-application-and-environment-am"></a> AM
+
+    Scripting application in AM could be summarized into the following categories:
+
+    * Authentication
+
+        * Modules and Chains
+            * Client-side
+            * Server-side
+        * Trees
+            * Client-side
+            * Server-side
+
+    * Authorization
+
+        * Scripting Policy Condition
+        * Access Token Modification
+
+    * Federation
+
+        * OIDC Claims Handling
+
+    #### <a id="am-scripting-client-side"></a>AM > Client-Side Scripts
+
+    AM may be used for authentication in the front channel. This can be assisted with custom client-side scripts written in JavaScript and executed in the user agent.
+
+    The use case for a client-side script is collecting information about the user agent's properties and its environment: [Geolocation](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/geolocation), IP, and whatever else that could be collected with a custom script running in a browser. Thus, the script needs to be written in JavaScript compatible with the browser, has access to the browser properties, and can make requests to external network resources _from the browser_. It could be used, for example, for detecting the client's IP.
+
+    The data collected by a client-side script can be submitted to the server side and become available for the server components involved in the same authentication procedure.
+
+    Scriptable access to the browser environment seems to be a unique feature of authentication procedures in AM, in comparison to the other script applications in the three products.
+
+    #### <a id="am-scripting-server-side"></a>AM > Server-Side Scripts
+
+    The decision making process on user identification and access management can be aided with the server-side scripts. The server-side scripts can be written in Groovy or JavaScript running on Rhino.
+
+    [Global Scripting API Functionality](https://backstage.forgerock.com/docs/am/6.5/dev-guide/#scripting-api-global) describes objects universally available for server-side scripts in AM. It also covers logging options, which is the only means of debugging server-side scripts in AM.
+
+    All server-side scripts have access to the following functionality:
+
+    * [Accessing HTTP Services](https://backstage.forgerock.com/docs/am/6.5/dev-guide/#scripting-api-global-http-client) with the client object, httpClient, and the `org.forgerock.http.protocol` package.
+
+        From scripts, AM makes synchronous, blocking network requests with the HTTP client object that are blocking until the script returns or times out according to the Server-side Script Timeout setting, which could be in the AM console under Configure > Global Services > Scripting > Secondary Configurations > AUTHENTICATION_SERVER_SIDE > Secondary Configurations > EngineConfiguration.
```

Comment:
This sentence definitely needs to be re-tuned :)

Reply:
Done!